rfd/0098-registered-openssh-nodes.md

```diff
@@ -0,0 +1,115 @@
+
+#### Option 1
+
+Each registered OpenSSH node will have a unique Agentless CA. The CA public key will include information about the node that it was created for, including the unique node name and the node's labels. When `(lib/srv.AuthHandlers).UserKeyAuth` is called to authenticate a node's user certificate, it will check if node information is present in the CA public key. If node information is present, an RBAC check will be preformed using the label information in the CA public key.
+
+Pros:
+
+- No need to lookup node resources to preform RBAC checks
+
+Cons:
+
+- Requires connecting to node to preform RBAC check
+- Updating node labels requires generating and distributing a new host key
+- CA rotation is more complex as every registered OpenSSH node requires a unique CA
+
+#### Option 2
+
+When a user sends a request to a Proxy to connect to a node, the Proxy will attempt to find a node resource by either its hostname or address, whichever the user specified. If the resource exists and has the `agentless` `sub_kind`, an RBAC check will be preformed. If the resource does not exist or isn't an `agentless` node, the connection flow will continue as normal.
+
+Pros:
+
+- No need to connect to node to preform RBAC check
+- Updating node labels can simply be done in DB
+- CA rotation is simple as every registered OpenSSH node will have the same CA
+
+Cons:
+
+- Node resources must be looked up before every node connection
+
+#### My choice: Option 2
+
+Option 2 makes CA rotation and updating node labels much easier, and has the only downside of having to search for node resources before connecting to one. 
+
+### Security
+
+Both RBAC checks and session recording for registered OpenSSH nodes require that users connect through a Proxy. If users are able to connect to registered OpenSSH nodes directly, they can bypass both features. Currently the User CA public key is copied to OpenSSH nodes and configured to be trusted by `sshd`. This means OpenSSH nodes will accept certificates that are signed by the User CA. The problem with that is when Teleport users authenticate with a cluster, the Auth server replies with a certificate that is signed by the User CA. Users could potentially use this certificate to directly connect to registered OpenSSH nodes.
+
+Using the new Agentless CA to sign certificates used to authenticate with registered OpenSSH nodes solves this problem, as Teleport users do not have access to any certificates signed by the Agentless CA.
+
+### UX
+
+The following Teleport node features won't work with registered OpenSSH nodes:
+
+- Enhanced session recording and restricted networking
+- Host user provisioning
+- Session recording without SSH session termination
+- Dynamic labels
+- Outbound persistent tunnels to Proxies
+
+Due to this and other potential future differences, `tsh ls` and the node listing on the web UI should be updated to display if nodes are registered OpenSSH nodes, or simply 'agentless'.
```

Comment:
Teleport currently doesn't support registering OpenSSH nodes. The only way to use OpenSSH nodes is to use proxy recording mode. I think that the only distinction that we can make is to show whether the node is a Teleport node or an agentless node.

Reply:
Oops, that's a typo, thanks